File: ina_2sem/labor_cs/3_lab/2_task/Romb.cpp
#include "Romb.h"

Romb::Romb(double bok, double kat) 
    : Czworokat(bok, bok, bok, bok, kat) {
    if (kat == 90) throw std::invalid_argument("Romb nie może mieć kąta 90°.");
}

double Romb::obliczPole() {
    return bok1 * bok1 * sin(kat * M_PI / 180.0);
}

double Romb::obliczObwod() {
    return 4 * bok1;
}

std::string Romb::podajNazwe() {
    return "Romb";
}

File: ina_2sem/labor_cs/3_lab/1_task/Romb.java
public class Romb extends Czworokat {
    public Romb(double bok, double kat) {
        super(bok, bok, bok, bok, kat);
        if (kat == 90) throw new IllegalArgumentException("Romb nie może mieć kąta 90°.");
    }

    @Override
    public double obliczPole() {
        return bok1 * bok1 * Math.sin(Math.toRadians(kat));
    }

    @Override
    public double obliczObwod() {
        return 4 * bok1;
    }

    @Override
    public String podajNazwe() {
        return "Romb";
    }
}
